server/server.go

```diff
@@ -62,3 +67,63 @@ func (s *Server) RenderingRegistrationDetails(w http.ResponseWriter, r *http.Req
 		return
 	}
 }
+
+func HomePageHandler(w http.ResponseWriter, r *http.Request) {
+	tmpl, _ := template.ParseFiles("views/homepage.html")
+	err := tmpl.Execute(w, nil)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func (s *Server) LoginFormHandler(w http.ResponseWriter, r *http.Request) {
+	sessionID := s.app.SessionService.GetSessionIDFromCookie(r)
+
+	if sessionID != 0 {
```

Comment:
This is a bit of a faulty check. I can modify the request to add a session cookie with any value and you would just redirect me to the homepage. This doesn't imply that I have a correct session or that it is active. You should check the session ID in the sessions table to verify that the session is active (it's there and it's not expired).